src/pyaro_readers/merging_reader.py

```diff
@@ -27,13 +27,23 @@ def __init__(self, data: list[Data], variable: str) -> None:
 
     @property
     def units(self) -> str:
-        base_unit = self._data[0].units
-        unit = cf_units.Unit(base_unit)
-        for d in self._data[1:]:
+        base_unit = None
+        for d in self._data:
+            if len(d) == 0:
+                continue
+            if base_unit is None:
+                base_unit = d.units
+                continue
+
+            unit = cf_units.Unit(base_unit)
             if unit.convert(1, d.units) != 1.0:
                 raise MergingReaderException(
                     f"The units are not the same in all the datasets {base_unit} {d.units}"
                 )
+        if base_unit is None:
```

Comment:
base_unit may be None if all datasets are None. This case should not throw an Exception.

Reply:
What should a reasonable default be for this case?

Reply:
Punting this to the first dataset might be better, so trying this